### PR TITLE
Normalise complement types in docmap generation

### PR DESCRIPTION
**Pull request by nlisgo**
part of https://github.com/elifesciences/data-hub-issues/issues/921

### CODE REVIEW

File: data_hub_api/docmaps/v2/codecs/elife_manuscript.py [hunk 1]
```diff
@@ -104,16 +105,20 @@ def iter_elife_manuscript_part_of_section_complement_for_one_record(
 ) -> Iterable[DocmapPartOfComplement]:
     if related_content['manuscript_id']:
         assert related_content['manuscript_type']
+        manuscript_type = related_content['manuscript_type'].lower().strip()
+        cleaned_manuscript_type = re.sub(r'[^a-z0-9 ]', ' ', manuscript_type)
+        manuscript_type_parts = re.split(r'\s+', cleaned_manuscript_type)
+        capitalized_parts = [part.capitalize() for part in manuscript_type_parts[1:]]
         yield {
-            'type': related_content['manuscript_type'],
+            'type': manuscript_type_parts[0] + ''.join(capitalized_parts),
```
Comment (nlisgo) [inline data_hub_api/docmaps/v2/codecs/elife_manuscript.py]: I was considering making this a util function with it's own tests but decided to just get the overall code working first. I can do that refactor if requested. Although there is no use for the code outside of this function.

File: data_hub_api/docmaps/v2/codecs/elife_manuscript.py [hunk 2]
```diff
@@ -104,16 +105,20 @@ def iter_elife_manuscript_part_of_section_complement_for_one_record(
 ) -> Iterable[DocmapPartOfComplement]:
     if related_content['manuscript_id']:
         assert related_content['manuscript_type']
+        manuscript_type = related_content['manuscript_type'].lower().strip()
+        cleaned_manuscript_type = re.sub(r'[^a-z0-9 ]', ' ', manuscript_type)
```
Comment (HazalCiplak) [inline data_hub_api/docmaps/v2/codecs/elife_manuscript.py]: Maybe we don't need this cleaning step as the manuscript type wouldn't include any unnecessary characters. We can just remove this part and update the test data instead something like "manuscript type 1". Let me know what do you think.

Comment (HazalCiplak) [reply data_hub_api/docmaps/v2/codecs/elife_manuscript.py]: Maybe can be shorten like:
```python
type_parts = [word.capitalize() for word in re.findall(r'\w+', related_content['manuscript_type'])]
type_name = ''.join(type_parts)
type_name = type_name[0].lower() + type_name[1:]
```

Comment (nlisgo) [reply data_hub_api/docmaps/v2/codecs/elife_manuscript.py]: Have made adjustments

**Review by HazalCiplak — APPROVED** [review]

**Pull request merged**